### PR TITLE
Update temperature gauge min / max / step values

### LINKED ISSUE #1511: Use different min/max/step values for temperature gauges depending on the temperature input type
As per slack thread [here](https://victrondevelopment.slack.com/archives/C020QG3JVEW/p1726867380981039), use different min/max/step values in the temperature gauges depending on the temperature type.

Currentl types for `/TemperatureType`:

```
0 - Battery
1 - Fridge
2 -Generic
```

Extend this list to: (min, max, step)

```
0 - Battery (-20, 60, 10)
1 - Fridge (0, 20, 5)
2 - Generic (-40, 60, 10)
3 - Room (5, 35, 10)
4 - Outdoor (-40, 60, 10)
5 - Water heater (0, 100, 10)
6 - Freezer (-30, 0, 5)
```

### PR DESCRIPTION
**Pull request by DanielMcInnes**
Fixes #1511

### CODE REVIEW

**Review by blammit — COMMENTED** [review]
Looks great. Could you update `LevelsPageConfig` to show various types of temperature sensors? There's some automatic handling there for setting properties on the temperature sensors. For example, changing:

```
{
            name: "Double gauge",
            inputs: [ { customName: "Refrigerator", temperature: 4.4223, humidity: 32.6075 } ]
        },
```

to

```
{
            name: "Double gauge",
            inputs: [ { temperatureType: VenusOS.Temperature_DeviceType_Freezer, temperature: -4.4223, humidity: 32.6075 } ]
},
```

seems to work fine. I think we should can rid of the "customName" setting for all of those dummy environment inputs anyway.

**Comment by DanielMcInnes** [comment]
> temperatureType: VenusOS.Temperature_DeviceType_Freezer

Thanks Bea, updated as suggested

File: data/EnvironmentInputs.qml [hunk 1]
```diff
@@ -41,5 +70,34 @@ QtObject {
...
+	}
+
+	function temperatureGaugeStepSize(temperatureType) {
+		return _validTemperatureType(temperatureType)
+				? (_temperatureDetails.get(temperatureType).temperatureGaugeStep)
+				: (_temperatureDetails.get(VenusOS.Temperature_DeviceType_Generic).temperatureGaugeStep)
+	}
+
+	function _validTemperatureType(temperatureType) {
+		return ((temperatureType >= 0) && (temperatureType < _temperatureDetails.count))
```
Comment (blammit) [inline data/EnvironmentInputs.qml:99]: Don't need the outer-most set of parenthesis?

**Review by blammit — APPROVED** [review]
Minor thing, I'm seeing this when changing UI configurations in the Levels page in mock mode:

```
Changing an uid of an item with valid value set is weird, ignoring value QVariant(QString, "Generic temperature sensor") for "mock/com.victronenergy.temperature.ttyUSB0/CustomName"
```

Otherwise LGTM.

**Comment by DanielMcInnes** [comment]
> Minor thing, I'm seeing this when changing UI configurations in the Levels page in mock mode:
> 
> ```
> Changing an uid of an item with valid value set is weird, ignoring value QVariant(QString, "Generic temperature sensor") for "mock/com.victronenergy.temperature.ttyUSB0/CustomName"
> ```
> 
> Otherwise LGTM.

Found it, fixed

File: data/EnvironmentInputs.qml (hunk 1 above)
Comment (DanielMcInnes) [reply data/EnvironmentInputs.qml]: Done

**Pull request merged**